click/types.py

```diff
@@ -119,6 +119,8 @@ def convert(self, value, param, ctx):
                         value = value.decode(fs_enc)
                     except UnicodeError:
                         value = value.decode('utf-8', 'replace')
+                else:
+                    value = value.decode('utf-8', 'replace')
```

Comment:
Please, Add TC :)

Reply:
Working on it. I'll aim to push a test case next week.